### PR TITLE
Nullable: Interlocked, Volatile

### PR DESCRIPTION
**Pull request by stephentoub**
Also fixing the class? annotation on public LazyInitializer methods.

### CODE REVIEW

File: src/System.Private.CoreLib/src/System/Threading/Interlocked.cs [hunk 1]
```diff
@@ -59,17 +60,17 @@ public static long Decrement(ref long location)
         public static extern double Exchange(ref double location1, double value);
 
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        public static extern object Exchange(ref object location1, object value);
+        public static extern object? Exchange(ref object? location1, object? value);
```
Comment (safern) [inline src/System.Private.CoreLib/src/System/Threading/Interlocked.cs]: Why is location1 nullable? I see that the native implementation throws if location1 == `NULL`

https://github.com/dotnet/coreclr/blob/4932d0f6fa87b9a6e3293aacd35cecedde44d87a/src/vm/comutilnative.cpp#L1863

File: src/System.Private.CoreLib/src/System/Threading/Interlocked.cs [hunk 2]
```diff
@@ -88,7 +89,7 @@ public static long Decrement(ref long location)
         public static extern double CompareExchange(ref double location1, double value, double comparand);
 
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        public static extern object CompareExchange(ref object location1, object value, object comparand);
+        public static extern object? CompareExchange(ref object? location1, object? value, object? comparand);
```
Comment (safern) [inline src/System.Private.CoreLib/src/System/Threading/Interlocked.cs]: Ditto. https://github.com/dotnet/coreclr/blob/4932d0f6fa87b9a6e3293aacd35cecedde44d87a/src/vm/comutilnative.cpp#L1880

File: src/System.Private.CoreLib/src/System/Threading/Interlocked.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/src/System/Threading/Interlocked.cs]: It's not the ref that's nullable, it's the thing the ref references that's nullable.

File: src/System.Private.CoreLib/src/System/Threading/Interlocked.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/src/System/Threading/Interlocked.cs]: Ditto

File: src/System.Private.CoreLib/src/System/Threading/Interlocked.cs (hunk 1 above)
Comment (safern) [reply src/System.Private.CoreLib/src/System/Threading/Interlocked.cs]: Ah right...

**Review by safern — APPROVED** [review]

**Pull request merged**